src/test_l2_dsCompositeIn.c

```diff
@@ -1,119 +1,307 @@
+{
+    UT_LOG("Entering test_l2_dsCompositeIn_VerifyCompositeInputStatus...");
+
+    dsError_t ret;
+    dsCompositeInStatus_t status;
+
+    // Step 1: Call dsCompositeInInit
+    ret = dsCompositeInInit();
+    UT_LOG("Invoked dsCompositeInInit(). Return status: %d", ret);
+    UT_ASSERT_EQUAL(ret, dsERR_NONE);
+
+    if (ret == dsERR_NONE)
+    {
+        // Step 2: Call dsCompositeInGetStatus
+        ret = dsCompositeInGetStatus(&status);
+        UT_LOG("Invoked dsCompositeInGetStatus(). Return status: %d", ret);
+        UT_ASSERT_EQUAL(ret, dsERR_NONE);
+
+        if (ret == dsERR_NONE)
+        {
+            // Step 3: Check 'isPresented' field
+            UT_LOG("'isPresented' field is: %d", status.isPresented);
+            UT_ASSERT_EQUAL(status.isPresented, false);
+
+            // Step 4: Check 'isPortConnected' field
+            for (int i = 0; i < dsCOMPOSITE_IN_PORT_MAX; i++)
+            {
+                UT_LOG("'isPortConnected' field for port %d is: %d", i, status.isPortConnected[i]);
+                UT_ASSERT_EQUAL(status.isPortConnected[i], false);
+            }
+
+            // Step 5: Check 'activePort' field
+            UT_LOG("'activePort' field is: %d", status.activePort);
+            UT_ASSERT_EQUAL(status.activePort, dsCOMPOSITE_IN_PORT_NONE);
+        }
+
+        // Step 6: Call dsCompositeInTerm
+        ret = dsCompositeInTerm();
+        UT_LOG("Invoked dsCompositeInTerm(). Return status: %d", ret);
+        UT_ASSERT_EQUAL(ret, dsERR_NONE);
+    }
+
+    UT_LOG("Exiting test_l2_dsCompositeIn_VerifyCompositeInputStatus...");
+}
 
 /**
- * @file test_l2_dsCompositeIn.c
- *
- */
+* @brief Test for setting and checking composite input port
+*
+* This function tests the APIs dsCompositeInInit, dsCompositeInSelectPort, dsCompositeInGetStatus, and dsCompositeInTerm. It first initializes the COMPOSITE Input sub-system, then selects a random port for presentation, gets the status of all COMPOSITE Input Status, checks the returned status, and finally terminates the COMPOSITE Input sub-system.
+*
+* **Test Group ID:** 02@n
+* **Test Case ID:** 003@n
+*
+* **Test Procedure:**
+* Refer to UT specification documentation [l2_dsCompositeIn_test_specification.md](l2_dsCompositeIn_test_specification.md)
+*/
 
-#include <string.h>
-#include <stdlib.h>
+void test_l2_dsCompositeIn_SetAndCheckCompositeInputPort(void)
+{
+    UT_LOG("Entering test_l2_dsCompositeIn_SetAndCheckCompositeInputPort...");
 
-#include <ut.h>
-#include <ut_log.h>
+    dsError_t ret;
+    dsCompositeInStatus_t status;
+    dsCompositeInPort_t port;
+
+    // Initialize the COMPOSITE Input sub-system
+    ret = dsCompositeInInit();
+    UT_ASSERT_EQUAL(ret, dsERR_NONE);
+    if (ret != dsERR_NONE)
+    {
+        UT_LOG("dsCompositeInInit failed with error: %d", ret);
+        return;
+    }
+
+    // Select a random port from the available ports
+    srand(time(NULL));
+    port = rand() % dsCOMPOSITE_IN_PORT_MAX;
```

Comment:
dsCompositeInGetNumberOfInputs needs to be called to get the supported composite in ports and use that value to loop all the ports. Spec is missing this information. Spec needs to be updated

Reply:
Done